### ISSUE TITLE
ListBox.SelectedIndexCollection modification works but IList versions throw NotSupportedException

### ISSUE THREAD

**Issue by hughbe** [dotnet/winforms#3087]
## Example Problems
E.g.
```cs
void IList.Clear()
{
    throw new NotSupportedException(SR.ListBoxSelectedIndexCollectionIsReadOnly);
}

public void Clear()
{
    if (owner != null)
    {
        owner.ClearSelected();
    }
}
```

Also, the properties `IsFixedSize` and `IsReadOnly` return `true`, but this is quite clearly not the case

## Proposed Changes
- Modify `IsFixedSize` and `IsReadOnly` to `false`
- Modify the IList modification methods to call the non-IList methods on the collection: `Add`, `Clear`, `Remove`

Thoughts? This would be breaking, but in the way that is good as we change exception to something functional: see https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/docs/coding-guidelines/breaking-changes.md

**Comment by hughbe** [comment]
Separately, I suggest changing the constructor to throw `ANE` for a null `owner`. This would ensure we don't get NREs calling methods. We made similar changes for `Control.ControlCollection`

**Comment by hughbe** [comment]
If we like this, will implement after #3086

**Comment by RussKie** [comment]
> Separately, I suggest changing the constructor to throw `ANE` for a null `owner`. This would ensure we don't get NREs calling methods. We made similar changes for `Control.ControlCollection`

Yes, we can do that (in a separate PR)

**Comment by RussKie** [comment]
> ## Proposed Changes
> * Modify `IsFixedSize` and `IsReadOnly` to `false`
> * Modify the IList modification methods to call the non-IList methods on the collection: `Add`, `Clear`, `Remove`

I think this is sensible. 
.NET Framework code appears to have the same, so it isn't new... 😕 

@dotnet/dotnet-winforms thoughts?

**Comment by merriemcgaw** [comment]
Seems sensible to me. What are the chances we break someone who was working around this issue in their Framework app?

**Comment by hughbe** [comment]
We're changing an exception into a success. I'd expect one of the circumstances that would change therefore is
```cs
try
{
   collection.Add(...);
}
catch
{
    return; // Or handle it.
}

// Do something else.
```
I honestly wouldn't expect this to be very common

**Comment by dreddy-work** [comment]
Thanks @hughbe for this proposal. Given this has been the behavior for long and could break migrating applications, we like to wait until there is a real customer scenario here.
